### PR TITLE
fix: added optional sign to `ParserOptions`'s jsdoc

### LINKED ISSUE #354: Setting `applyTraits` on Typescript projects
#### Describe the bug
I should be able to set `applyTraits` option in parser method without providing other options in Typescript projects. But since all the parameters are mandatory in `ParserOptions` type, the compiler will complain about specifying all of them.

#### How to Reproduce
Call the `parser` function with `{applyTraits: false}` option in a Typescript project.
you will get the following error: 
`Type '{ applyTraits: false; }' is missing the following properties from type 'ParserOptions': path, parse, resolve`

#### Expected behavior
The parser should parse the document without any errors.

### PR DESCRIPTION
**Pull request by KhudaDad414**
**Description**
makes parser options optional for Typescript projects.
**Related issue(s)**
Resolves #354

### CODE REVIEW

File: API.md [hunk 1]
```diff
@@ -3200,8 +3200,7 @@ Implements functions to deal with the AsyncAPI document.
 <a name="module_@asyncapi/parser+AsyncAPIDocument+traverseSchemas"></a>
 
 #### asyncAPIDocument.traverseSchemas(callback, schemaTypesToIterate)
-Traverse schemas in the document and select which types of schemas to include.
-By default all schemas are iterated
+Traverse schemas in the document and select which types of schemas to include.By default all schemas are iterated
```
Comment (jonaslagoni) [inline API.md:3203]: Where does this change come from? 🤔

File: lib/parser.js [hunk 2]
```diff
@@ -30,10 +30,10 @@ module.exports = {
...
  * @typedef {Object} ParserOptions
- * @property {String} path - Path to the AsyncAPI document. It will be used to resolve relative references. Defaults to current working dir.
- * @property {Object} parse - Options object to pass to {@link https://apidevtools.org/json-schema-ref-parser/docs/options.html|json-schema-ref-parser}.
- * @property {Object} resolve - Options object to pass to {@link https://apidevtools.org/json-schema-ref-parser/docs/options.html|json-schema-ref-parser}.
- * @property {Boolean} applyTraits - Whether to resolve and apply traits or not. Defaults to true.
+ * @property {String=} path - Path to the AsyncAPI document. It will be used to resolve relative references. Defaults to current working dir.
+ * @property {Object=} parse - Options object to pass to {@link https://apidevtools.org/json-schema-ref-parser/docs/options.html|json-schema-ref-parser}.
+ * @property {Object=} resolve - Options object to pass to {@link https://apidevtools.org/json-schema-ref-parser/docs/options.html|json-schema-ref-parser}.
+ * @property {Boolean=} applyTraits - Whether to resolve and apply traits or not. Defaults to true.
```
Comment (jonaslagoni) [inline lib/parser.js:36]: This is actually all you need to commit, the rest will be auto-generated by the CI system. This way we are sure no collisions occur. Please revert the changes in `types.d.ts` and `API.md`

**Review by jonaslagoni — CHANGES_REQUESTED** [review]

**Comment by jonaslagoni** [comment]
@KhudaDad414 thanks for the fast fix!

Can you rename the title of the pr to include `fix: ...`? 🙂

File: API.md (hunk 1 above)
Comment (KhudaDad414) [reply API.md:3203]: Don't know man. searching for it. I think it's some kind of newline conversion thing.

File: lib/parser.js (hunk 2 above)
Comment (KhudaDad414) [reply lib/parser.js:36]: @jonaslagoni thanks for the quick review. :) it's ready.

**Review by jonaslagoni — APPROVED** [review]
👍

**Pull request merged**